Sources/RegexBenchmark/Suite/Backtracking.swift

```diff
@@ -0,0 +1,45 @@
+import _StringProcessing
+import RegexBuilder
+import Foundation
+
+// Tests that involve heavy backtracking
+
+extension BenchmarkRunner {
+  mutating func addBacktracking() {
+    let r = "^ +A"
+    let s = String(repeating: " ", count: 10000)
+
+    let basicBacktrack = Benchmark(
+      name: "BasicBacktrack",
+      regex: try! Regex(r),
+      ty: .allMatches,
+      target: s
+    )
+
+    let basicBacktrackNS = NSBenchmark(
+      name: "BasicBacktrackNS",
+      regex: try! NSRegularExpression(pattern: r),
+      ty: .all,
+      target: s
+    )
+
+    let basicBacktrackFirstMatch = Benchmark(
+      name: "BasicBacktrackFirstMatch",
+      regex: try! Regex(r),
+      ty: .first,
+      target: s
+    )
+
+    let basicBacktrackNSFirstMatch = NSBenchmark(
+      name: "BasicBacktrackNSFirstMatch",
+      regex: try! NSRegularExpression(pattern: r),
+      ty: .first,
+      target: s
+    )
+    
```

Comment:
Hmm... we want whole-match benchmarks, match-from-front benchmarks, first-match, and all-matches (which is repeated first-match calls). We also want to make the NSRegularExpression equivalent of each, at least if there can be an equivalent.